fedimint-server/src/config/api.rs

```diff
@@ -378,6 +399,86 @@ impl ConfigGenApi {
     fn bad_request<T>(msg: &str) -> ApiResult<T> {
         Err(ApiError::bad_request(msg.to_string()))
     }
+
+    pub async fn restart_federation_setup(&self) -> ApiResult<()> {
+        let leader = {
+            let expected_status = [
+                ServerStatus::SharingConfigGenParams,
+                ServerStatus::ReadyForConfigGen,
+                ServerStatus::ConfigGenFailed,
+                ServerStatus::VerifyingConfigs,
+                ServerStatus::VerifiedConfigs,
+            ];
+            let mut state = self.require_status_maybe(&expected_status).map_err(|_| {
+                ApiError::bad_request(format!(
+                    "Expected to be in one of {:?} states",
+                    expected_status
+                ))
+            })?;
+
+            self.remove_configs()?;
+            state.status = ServerStatus::SetupRestarted;
+            info!(
+                target: fedimint_logging::LOG_NET_PEER_DKG,
+                "Update config gen status to 'Setup restarted'"
+            );
+            // Create a WSClient for the leader
```

Comment:
Why not just set the state to like `AwaitingPassword` or whatever the initial state is and return right here?

Reply:
Makes it much harder for UX to determine clear intent to restart setup. We'd need to add an additional info to `AwaitingPassword` to differentiate first time setup vs restart of setup. Alternatively, `SetupRestarted` is a clear state to signal request to restart by one of the peers, and we can watch it to see which peers have accepted the restart